src/components/VideoListingCard.js

```diff
@@ -0,0 +1,43 @@
+import { Link } from "react-router-dom"
+import { useState } from "react"
+
+const VideoListingCard = ({ video }) => {
+
+    const [videoOptions, setVideoOptions] = useState(false);
+
+    const toggleVideoOptions = () => setVideoOptions(state => !state);
```

Comment:
just a suggestion, for this i think you can also do setVideoOptions(!videoOptions),